L.Control.Geonames.js

```diff
@@ -91,6 +100,50 @@ L.Control.Geonames = L.Control.extend({
             callbackName
         );
     },
+    _getSearchParams: function() {
+        var params = '';
+
+        for (var paramName in this.options.adminCodes) {
+            if (ADMIN_CODES.indexOf(paramName) == -1) {
+                continue;
+            }
+
+            var paramValue = this.options.adminCodes[paramName];
+
+            if (typeof paramValue == 'function') {
+                paramValue = paramValue();
+                if (paramValue) {
+                    params += '&' + paramName + '=' + paramValue;
+                }
+            } else if (typeof paramValue == 'string' && paramValue.length > 0) {
+                params += '&' + paramName + '=' + paramValue;
+            }
+        }
+
+        var bbox = this._getBbox();
+        if (bbox) {
+            params += bbox;
+        }
+
+        return params;
+    },
+    _getBbox: function() {
+        if (this._bbox) {
+            return this._bbox;
+        }
+
+        var bbox = '';
+        for (var i in BBOX) {
+            var dir = BBOX[i];
+            if (!this.options.bbox[dir]) {
+                bbox = '';
+                break;
```

Comment:
@ka7eh This should either return from the function with an empty string, or raise an exception.  Otherwise, you are just dropping parameters that aren't valid.